com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java

```diff
@@ -27,9 +35,47 @@
 import com.dynamo.bob.pipeline.LuaScanner.Property.Status;
 import com.dynamo.bob.util.MurmurHash;
 import com.dynamo.gameobject.proto.GameObject.PropertyType;
-import com.dynamo.bob.pipeline.DefoldLuaParser.PropertyAndLine;
+import com.dynamo.bob.pipeline.antlr.LuaParser;
+import com.dynamo.bob.pipeline.antlr.LuaLexer;
+import com.dynamo.bob.pipeline.antlr.LuaParserBaseListener;
+
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
```

Comment:
Why do we have some code in an external jar, and some code in a local (copied) package (now renamed to antlr)?

Reply:
Antlr is the generic tool/framework we use to parse the Lua code (https://www.antlr.org/). It contains one part that is the parser (the jar) and one part which is the generated parser of Lua code (the java files). This is the way Antlr works by generating a parser based on a language definition. The grammar of the language is defined in the .g4 files in the same package path. This means that Antlr can work on any structured language as long as there is a language definition for it. And there is a huge list of language grammars on GitHub: https://github.com/antlr/grammars-v4